widgets/detector_settings.py

```diff
@@ -519,14 +522,45 @@ def __add_efficiency(self):
                 QtWidgets.QMessageBox.critical(
                     self, "Error",
                     f"There already is an efficiency file for element "
-                    f"{used_eff_file.stem}.\n",
+                    f"{current_eff_file.stem}.\n",
                     QtWidgets.QMessageBox.Ok, QtWidgets.QMessageBox.Ok)
 
         self.efficiency_files = self.obj.get_efficiency_files()
         # Store the folder where we previously fetched an eff-file
-        gutils.set_potku_setting(
-            DetectorSettingsWidget.EFF_FILE_FOLDER_KEY,
-            str(eff_file.parent))
+        gutils.set_potku_setting(DetectorSettingsWidget.EFF_FILE_FOLDER_KEY,
+                                 str(eff_file.parent))
+
+    def __check_chosen_elements(self, cuts_list, current_eff_file,
+                                current_eff_files, eff_file):
+        found_eff_files = []
+        for cut in cuts_list:  # Checks if the selected element
+            # has a cut file
+            for c in cut:
+                cut_element_str = c.name.split(".")[1]
+                # Writes down if a corresponding element is found
+                if cut_element_str in current_eff_file.stem:
+                    found_eff_files.append(cut_element_str)
+        for file in found_eff_files:
+            if current_eff_file.stem not in file:  # Gives a warning
+                # if a cut file is not found
+                reply = QtWidgets.QMessageBox.warning(
+                    self, "Warning",
+                    f"All measurements do not have a "
+                    f"cut file for the element "
+                    f"{eff_file.stem}.\nDo you want to "
+                    f"continue?",
```

Comment:
The wording on this warning is a bit ambiguous. Does it mean
1. "None of the measurements have a cut file for the element {eff_file.stem}.\nDo you want to continue?" or
2. "At least one of the measurements is missing a cut file for the element {eff_file.stem}\nDo You want to continue?"?